src/StockDetails/WatchlistPickerPopup.jsx

```diff
@@ -23,7 +23,10 @@ const WatchlistPickerPopup = ({ visible, onClose, symbol }) => {
       onCancel={onClose}
       confirmButtonText="OK"
       onConfirm={() => {
-        if (selectedWatchlist) dispatch(addStockToWatchlist(selectedWatchlist, symbol));
+        if (selectedWatchlist) {
+          dispatch(addStockToWatchlist(selectedWatchlist, symbol));
+          dispatch(fetchWatchlistDetail(symbol));
```

Comment:
I would dispatch this action in the addStockWatchlist action creator, otherwise we always need to call both of them (if we ccan add a watchlist from somewhere else then this component)